sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -218,15 +234,38 @@ function appendQueryParams(url: string, queryParams: Map<string, string | string
   // QUIRK: parsedUrl.searchParams will have their name/value pairs decoded, which
   // can change their meaning to the server, such as in the case of a SAS signature.
   // To avoid accidentally un-encoding a query param, we parse the key/values ourselves
-  const existingParams = simpleParseQueryParams(parsedUrl.search);
-  const combinedParams = new Map<string, string | string[]>(existingParams);
+  const combinedParams = simpleParseQueryParams(parsedUrl.search);
 
   for (const [name, value] of queryParams) {
     const existingValue = combinedParams.get(name);
     if (Array.isArray(existingValue)) {
-      existingValue.push(...value);
+      if (Array.isArray(value)) {
+        if (existingValue.toString() !== value.toString()) {
+          existingValue.push(...value);
+        }
+      } else {
+        existingValue.push(value);
+      }
     } else if (existingValue) {
-      combinedParams.set(name, [existingValue, ...value]);
+      if (operationSpec.queryParameters?.length) {
+        for (const queryParameter of operationSpec.queryParameters) {
+          if (queryParameter.mapper.serializedName === name) {
+            if (queryParameter.mapper.type.name === "Sequence") {
+              if (Array.isArray(value)) {
+                combinedParams.set(name, [existingValue, ...value]);
+              } else {
+                combinedParams.set(name, [existingValue, value]);
+              }
+            } else {
+              if (Array.isArray(value)) {
+                combinedParams.set(name, [existingValue, ...value]);
+              } else {
+                combinedParams.set(name, value);
+              }
+            }
+          }
+        }
+      }
```

Comment:
Based on my two previous comments I think the body of this `else` could end up looking something like

```suggestion
      if (Array.isArray(value)) {
          value.unshift(existingValue);
          combinedParams.set(name, value);
      } else if (sequenceParams.has(name)) {
          combinedParams.set(name, [existingValue, value]);
      } else {
          combinedParams.set(name, value);
      }
```

or maybe if we wanted to get a little more compact

```ts
      let newValue = value;
      if (Array.isArray(value)) {
          value.unshift(existingValue);
      } else if (sequenceParams.has(name)) {
          newValue = [existingValue, value];
      }
      combinedParams.set(name, newValue);
```

Reply:
Wonderful!!!!!! Applied it in the next commit.